nx-X11/programs/Xserver/hw/nxagent/Handlers.c

```diff
@@ -956,7 +956,20 @@ void nxagentShadowWakeupHandler(void * data, int count, void * mask)
 
 void nxagentHandleCollectInputFocusEvent(int resource)
 {
+  /*
+   * While we don't even need window or revert_to later on, a discrepancy in
+   * data type sizes between the X server (Window being a 32bit ID) and
+   * the Xlib (Window being a 64bit ID) will lead to stack corruption here.
+   * Calling functions from nxcompext from nxagent sounds like a very bad idea
+   * to begin with, but let's assume that's necessary for now and work around
+   * the corruption issue.
+   */
+#ifdef _XSERVER64
+  Window64 window;
+#else
   Window window;
+#endif /* defined (_XSERVER64) */
```

Comment:
Then the NX code is inconsistent again... I've been looking mostly at `nx-X11/programs/Xserver/hw/nxagent/Colormap.c`, which uses `_XSERVER64` and switches between `Window` and `Window64`.

For our use case in this commit, it's probably better to use `XlibWindow`, I agree. Sadly stuff of Xlib* types can't be passed down to other things within the X.Org Server code (exactly because the sizes might differ), but especially in this case the data is unused anyway.